### ISSUE TITLE
extend password_hashes test case to allow sssd

### ISSUE THREAD

**Issue by Vincinator** [gardenlinux/gardenlinux#1505]
**What would you like to be added**:
The [tests/helper/tests/password_hashes](https://github.com/gardenlinux/gardenlinux/blob/main/tests/helper/tests/password_hashes.py#L12) test looks for a line  in /etc/pam.d/common-password that contains "[success=1 default=ignore]".

Then 2 checks happen:
- only one line allowed containing *"[success=1 default=ignore]"
- yescrypt or sha512 must also be defined in that line

By a default non-modified `make metal` build , the `/etc/pam.d/common-password` file looks like: 
```
password        [success=1 default=ignore]      pam_unix.so obscure yescrypt
# here's the fallback if no module succeeds
password        requisite                       pam_deny.so
# prime the stack with a positive return value if there isn't one already;
# this avoids us returning an error just because nothing sets a success code
# since the modules above will each just jump around
password        required                        pam_permit.so
# and here are more per-package modules (the "Additional" block)
# end of pam-auth-update config
``` 

Installing sssd changes the `/etc/pam.d/common-password` file to:
```
password        [success=2 default=ignore]      pam_unix.so obscure use_authtok try_first_pass yescrypt
password        sufficient                      pam_sss.so use_authtok
# here's the fallback if no module succeeds
password        requisite
# prime the stack with a positive return value if there isn't one already;
# this avoids us returning an error just because nothing sets a success code
# since the modules above will each just jump around
password        required                        pam_permit.so
# and here are more per-package modules (the "Additional" block)
# end of pam-auth-update config
``` 

**Why is this needed**:
If a user decides to use `sssd`, the test should not fail if the required modules are configured.

**Comment by Vincinator** [comment]
tl;dr: What does sssd change in terms of Password modules? Adds `use_authtok` and  `try_first_pass`, and another fallback module named `pam_sss.so`.

Long:
If sssd is installed, the `/etc/pam.d/common-password` looks like this:
```
password        [success=2 default=ignore]      pam_unix.so obscure use_authtok try_first_pass yescrypt
password        sufficient                      pam_sss.so use_authtok
# here's the fallback if no module succeeds
password        requisite                       pam_deny.so
# prime the stack with a positive return value if there isn't one already;
# this avoids us returning an error just because nothing sets a success code
# since the modules above will each just jump around
password        required                        pam_permit.so
# and here are more per-package modules (the "Additional" block)
# end of pam-auth-update config
```
sssd tries to add `use_authtok` and  `try_first_pass` modules to the first line. I assume this should be no problem. 
The matching config in `/usr/share/pam-configs/` for the `pam_unix.so` module should be `/usr/share/pam-configs/unix`.

This `/usr/share/pam-configs/unix` exists independently of sssd, it is not changed by sssd: 
```
cat /usr/share/pam-configs/unix 
Name: Unix authentication
Default: yes
Priority: 256
Auth-Type: Primary
Auth:
        [success=end default=ignore]    pam_unix.so nullok try_first_pass
Auth-Initial:
        [success=end default=ignore]    pam_unix.so nullok
Account-Type: Primary
Account:
        [success=end new_authtok_reqd=done default=ignore]      pam_unix.so
Account-Initial:
        [success=end new_authtok_reqd=done default=ignore]      pam_unix.so
Session-Type: Additional
Session:
        required        pam_unix.so
Session-Initial:
        required        pam_unix.so
Password-Type: Primary
Password:
        [success=end default=ignore]    pam_unix.so obscure use_authtok try_first_pass yescrypt
Password-Initial:
        [success=end default=ignore]    pam_unix.so obscure yescrypt
```
